apps/web/src/components/templates/NotificationSettingsForm.tsx

```diff
@@ -1,56 +1,38 @@
 import { Controller, useFormContext } from 'react-hook-form';
-import { useMutation, useQuery } from 'react-query';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
 import { useEffect, useState } from 'react';
 import { Grid } from '@mantine/core';
 import { getNotificationGroups } from '../../api/notifications';
 import { api } from '../../api/api.client';
 import { Input, Select } from '../../design-system';
 
 export const NotificationSettingsForm = ({ editMode }: { editMode: boolean; errors: any }) => {
+  const queryClient = useQueryClient();
   const {
     formState: { errors },
     setValue,
     control,
   } = useFormContext();
-  const {
-    data: serverGroups,
-    isLoading: loadingGroups,
-    refetch: refetchGroups,
-  } = useQuery('notificationGroups', getNotificationGroups);
-
+  const { data: groups, isLoading: loadingGroups } = useQuery('notificationGroups', getNotificationGroups);
   const { isLoading: loadingCreateGroup, mutateAsync: createNotificationGroup } = useMutation<
     { name: string; _id: string },
     { error: string; message: string; statusCode: number },
     {
       name: string;
     }
-  >((data) => api.post(`/v1/notification-groups`, data));
-
-  const [groups, setGroups] = useState<{ name: string; _id: string }[]>([]);
-
-  useEffect(() => {
-    if (serverGroups) {
-      setGroups(serverGroups);
-      if (!editMode && serverGroups?.length) {
-        /**
-         * Quick fix to default value of notificationGroup not set when page is not refreshed
-         */
-        setTimeout(() => setValue('notificationGroup', serverGroups[0]._id), 0);
-      }
-    }
-  }, [serverGroups]);
+  >((data) => api.post(`/v1/notification-groups`, data), {
+    onSuccess: (data) => {
+      queryClient.setQueryData('notificationGroups', [...groups, data]);
```

Comment:
Instead of refetching the entire notifications group on creation of a new one we just update the cache with the newly created item

Reply:
Amazing💃🏼 Good to know